codalab/rest/block_schemas.py

```diff
@@ -0,0 +1,96 @@
+"""
+Marshmallow schemas for serializing resource dicts into JSON API documents, and vice-versa.
+The schemas also perform some basic validation.
+Placed here in this central location to avoid circular imports.
+"""
+from bottle import local
+from marshmallow import (
+    Schema as PlainSchema,
+    ValidationError,
+    validate,
+    validates_schema,
+)
+from marshmallow_jsonapi import Schema, fields
+
+
+# Enum that represents different modes for a block.
+class BlockModes:
+    MARKUP_MODE = "markup_block"
+    RECORD_MODE = "record_block"
+    TABLE_MODE = "table_block"
+    BUNDLE_CONTENTS_MODE = "contents_block"
+    BUNDLE_HTML_MODE = "html_block"
+    BUNDLE_IMAGE_MODE = "image_block"
+    GRAPH_MODE = "graph_block"
+    WSEARCH_MODE = "wsearch_block"
+    SEARCH_MODE = "search_block"
+
+    values = (MARKUP_MODE, RECORD_MODE, TABLE_MODE, BUNDLE_CONTENTS_MODE,
+              BUNDLE_HTML_MODE, BUNDLE_IMAGE_MODE, GRAPH_MODE, WSEARCH_MODE,
+              SEARCH_MODE)
+
+
+STATUS_STRINGS = ("unknown", "ready", "not_found", "no_permission")
```

Comment:
Need to think more carefully about whether this is the right set of status strings...

For example, if I don't have permission, should I get a 'not_found' or 'no_permission' (which reveals the existence of certain resources even if I don't have access)...

Reply:
I think it is okay to have a distinction between not_found and no_permission, since we already do that for resources in our current front-end.

Reply:
What is unknown? Just like a generic catch-all?

Reply:
Should there be a "fetching" status that represents something that is currently trying to be fetched?